src/sortStudents.ts

```diff
@@ -1,7 +1,94 @@
-// describe Student type
-// create and export SortType enum
-// create SortOrder type
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married:boolean;
+  grades: number[];
+}
+
+export enum SortType {
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): object {
+  const sortedStudents: object = [...students];
+
+  function avarageNum(numArray: number[]): number {
+    return numArray.reduce((sum, num) => sum + num, 0) / numArray.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        sortedStudents.sort((studentA: Student, studentB: Student) => {
+          return studentA.name.localeCompare(studentB.name);
+        });
+      } else {
+        sortedStudents.sort((studentA: Student, studentB: Student) => {
+          return studentB.name.localeCompare(studentA.name);
+        });
+      }
+      break;
```

Comment:
use sortBy instead of key names

```suggestion
    case SortType.Name:
      if (order === 'asc') {
        sortedStudents.sort((studentA: Student, studentB: Student) => {
          return studentA[sortBy].localeCompare(studentB[sortBy]);
        });
      } else {
        sortedStudents.sort((studentA: Student, studentB: Student) => {
          return studentB[sortBy].localeCompare(studentA[sortBy]);
        });
      }
      break;
```